src/test/java/co/rsk/bitcoinj/core/TransactionWitnessTest.java

```diff
@@ -361,10 +364,290 @@ public void getSigInsertionIndex_withWitnessFilledWithSignatures_shouldReturnThe
+        TransactionWitness witnessWithTwoSignatures = witnessWithFedKey2Signature.updateWitnessWithSignature(
+            p2shP2wshOutputScript, fedKey1SignatureEncoded, fed1SigInsertionIndex);
+
+        ArrayList<byte[]> expectedSignatures = Lists.newArrayList(fedKey1SignatureEncoded, fedKey2SignatureEncoded);
+        assertSignaturesAreInOrder(witnessWithTwoSignatures, expectedSignatures);
+    }
+
+    @Test
+    public void updateWitnessWithSignature_withThreeSignaturesInDescendingOrder_shouldBeInsertedRespectingTheOrder() {
+        // signing order: [fedKey3, fedKey2, fedKey1]
+        // arrange
+        byte[] fedKey3SignatureEncoded = getTransactionSignatureEncodedToBtc(fedKey3, btcTxSigHashForWitness);
+        TransactionWitness witnessWithoutSignatures = btcTx.getWitness(FIRST_INPUT_INDEX);
+
+        // act & assert
+        int fed3SigInsertionIndex = witnessWithoutSignatures.getSigInsertionIndex(btcTxSigHashForWitness, fedKey3);
+        TransactionWitness witnessWithFedKey3Signature = witnessWithoutSignatures.updateWitnessWithSignature(
+            p2shP2wshOutputScript, fedKey3SignatureEncoded, fed3SigInsertionIndex);
+
+        assertSignaturesAreInOrder(witnessWithFedKey3Signature, Lists.newArrayList(fedKey3SignatureEncoded));
+
+        byte[] fedKey2SignatureEncoded = getTransactionSignatureEncodedToBtc(fedKey2, btcTxSigHashForWitness);
+        int fed2SigInsertionIndex = witnessWithFedKey3Signature.getSigInsertionIndex(btcTxSigHashForWitness, fedKey2);
+        TransactionWitness witnessWithTwoSignatures = witnessWithFedKey3Signature.updateWitnessWithSignature(
+            p2shP2wshOutputScript, fedKey2SignatureEncoded, fed2SigInsertionIndex);
+
+        assertSignaturesAreInOrder(witnessWithTwoSignatures, Lists.newArrayList(fedKey2SignatureEncoded, fedKey3SignatureEncoded));
+
+        byte[] fedKey1SignatureEncoded = getTransactionSignatureEncodedToBtc(fedKey1, btcTxSigHashForWitness);
+        int fed1SigInsertionIndex = witnessWithTwoSignatures.getSigInsertionIndex(btcTxSigHashForWitness, fedKey1);
+        TransactionWitness witnessWithThreeSignatures = witnessWithTwoSignatures.updateWitnessWithSignature(
+            p2shP2wshOutputScript, fedKey1SignatureEncoded, fed1SigInsertionIndex);
+
+        assertSignaturesAreInOrder(witnessWithThreeSignatures, Lists.newArrayList(fedKey1SignatureEncoded, fedKey2SignatureEncoded, fedKey3SignatureEncoded));
+    }
+
+    @Test
+    public void updateWitnessWithSignature_withThreeUnorderedSignatures_shouldBeInsertedRespectingTheOrder() {
+        // signing order: [fedKey1, fedKey3, fedKey2]
+        // arrange
+        byte[] fedKey1SignatureEncoded = getTransactionSignatureEncodedToBtc(fedKey1, btcTxSigHashForWitness);
+        TransactionWitness witnessWithoutSignatures = btcTx.getWitness(FIRST_INPUT_INDEX);
+
+        // act & assert
+        int fed1SigInsertionIndex = witnessWithoutSignatures.getSigInsertionIndex(btcTxSigHashForWitness, fedKey1);
```

Comment:
To apply in all tests, let's assert the expected value of the sig insertion index. It gives a clearer idea of what's happening